source/helpers/render-semantic-release-commit-types.ts

```diff
@@ -0,0 +1,28 @@
+import zipTextNodes from 'zip-text-nodes';
+
+// Using https://github.com/angular/angular/blob/main/CONTRIBUTING.md?rgh-link-date=2024-09-09T05%3A51%3A31Z#-commit-message-format as the reference for the seamntic commit message format.
+
+export function getSemanticCommitAndScope(commitTitle: string): [string, string?] | undefined {
+	const match = /^(\w*)(?:\((.+)\))?!?:.*/.exec(commitTitle);
```

Comment:
Did ChatGPT’s regex not work? I had specifically requested named groups as well. Not that .* is meaningless at the end because we’re not capturing that part and the regex matches the first part just as well.

Reply:
Part of the regex is to detect if it is a valid semantic commit. "feat:" is not a valid semantic release commit and should not be matched by the regex. There may be other quirks to a valid semantic commit that I do not know. For now this regex should pass the tests.